packages/@n8n/nodes-langchain/nodes/chains/TextClassifier/TextClassifier.node.ts

```diff
@@ -0,0 +1,253 @@
+import type {
+	IDataObject,
+	IExecuteFunctions,
+	INodeExecutionData,
+	INodeParameters,
+	INodeType,
+	INodeTypeDescription,
+} from 'n8n-workflow';
+
+import { NodeConnectionType } from 'n8n-workflow';
+
+import type { BaseLanguageModel } from '@langchain/core/language_models/base';
+import { HumanMessage } from '@langchain/core/messages';
+import { SystemMessagePromptTemplate, ChatPromptTemplate } from '@langchain/core/prompts';
+import { StructuredOutputParser } from 'langchain/output_parsers';
+import { z } from 'zod';
+import { getTracingConfig } from '../../../utils/tracing';
+import { getPromptInputByType } from '../../../utils/helpers';
+
+const SYSTEM_PROMPT_TEMPLATE =
+	"Please classify the text provided by the user into one of the following categories: {categories}, and use the provided formatting instructions below. Don't explain, and only output the json.";
+
+const configuredOutputs = (parameters: INodeParameters) => {
+	const categories = ((parameters.categories as IDataObject)?.categories as IDataObject[]) ?? [];
+	const fallback = parameters?.fallback as boolean;
+	const ret = categories.map((cat) => {
+		return { type: NodeConnectionType.Main, displayName: cat.category };
+	});
+	if (fallback) ret.push({ type: NodeConnectionType.Main, displayName: 'Other' });
+	return ret;
+};
+
+export class TextClassifier implements INodeType {
+	description: INodeTypeDescription = {
+		displayName: 'Text Classifier',
+		name: 'textClassifier',
+		icon: 'fa:tags',
+		group: ['transform'],
+		version: 1,
+		description: 'Classify your text into distinct categories',
+		codex: {
+			categories: ['AI'],
+			subcategories: {
+				AI: ['Chains', 'Root Nodes'],
+			},
+			resources: {
+				primaryDocumentation: [
+					{
+						url: 'https://docs.n8n.io/integrations/builtin/cluster-nodes/root-nodes/n8n-nodes-langchain.chainllm/',
+					},
+				],
+			},
+		},
+		defaults: {
+			name: 'Text Classifier',
+		},
+		inputs: [
+			{ displayName: '', type: NodeConnectionType.Main },
+			{
+				displayName: 'Model',
+				maxConnections: 1,
+				type: NodeConnectionType.AiLanguageModel,
+				required: true,
+			},
+		],
+		outputs: `={{(${configuredOutputs})($parameter)}}`,
+		properties: [
+			{
+				displayName: 'Categories',
+				name: 'categories',
+				placeholder: 'Add Category',
+				type: 'fixedCollection',
+				default: {},
+				typeOptions: {
+					multipleValues: true,
+				},
+				options: [
+					{
+						name: 'categories',
+						displayName: 'Categories',
+						values: [
+							{
+								displayName: 'Category',
+								name: 'category',
+								type: 'string',
+								default: '',
+								description: 'Category to add',
+							},
+							{
+								displayName: 'Description',
+								name: 'description',
+								type: 'string',
+								default: '',
+								description: "Describe your category if it's not obvious",
+							},
```

Comment:
Neither of these fields is required; is that intentional? IMO at least `category` should be mandatory